### PR TITLE
Interpret the empty string as if the constraint is not specified

### PR DESCRIPTION
**Pull request by youennf**

### CODE REVIEW

**Comment by alvestrand** [comment]
For backwards compatibility reasons, it seems that we only want to accept "empty string" as a device ID, not for every constraint - other constraints have "MUST be a member of the following values..."

**Comment by youennf** [comment]
I am fine with special casing empty string to deviceId only.

> other constraints have "MUST be a member of the following values..."

Can you point me to the part of the spec requiring this?

File: getusermedia.html [hunk 1]
```diff
@@ -1642,6 +1642,9 @@ <h2>Constrainable Properties</h2>
                 "MediaStreamTrack">applyConstraints()</a>, since any attempt to
                 set a different value will result in an unsatisfiable
                 <a>ConstraintSet</a>.
+                If a string of length 0 is used as a deviceID value constraint
```
Comment (henbos) [inline getusermedia.html]: deviceId

**Comment by youennf** [comment]
Updated typo from deviceID to deviceId.

**Pull request merged**